lib/authorizer/hawk.js

```diff
@@ -225,30 +226,34 @@ module.exports = {
 
         request.removeHeader('Authorization', {ignoreCase: true});
 
-        computeBodyHash(request.body, params.algorithm, 'base64', contentType, function (bodyHash) {
-            result = self.computeHeader({
-                credentials: {
-                    id: params.authId,
-                    key: params.authKey,
-                    algorithm: params.algorithm
-                },
-                nonce: params.nonce,
-                timestamp: params.timestamp,
-                ext: params.extraData,
-                app: params.app,
-                dlg: params.delegation,
-                user: params.user,
-                url: url.href, // force toString to add a protocol to the URL.
-                method: request.method,
-                hash: bodyHash
+        computeBodyHash(params.includePayloadHash && request.body, // pass body only when payload hash is required
```

Comment:
this `&&` approach does not differentiate between blank body versus option to not hash body.

Reply:
@vikiCoder This will work but the intent is not clear here.
Instead, do an early bail out based on `includePayloadHash`.